pkg/api/apis/operators/v1alpha1/installplan_test.go

```diff
@@ -0,0 +1,61 @@
+package v1alpha1
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestOrderSteps(t *testing.T) {
+	tests := []struct {
+		description string
+		in          []*Step
+	}{
+		{
+			description: "EmptyList",
+			in:          []*Step{},
+		},
+		{
+			description: "csvsRDS",
+			in:          []*Step{step(crdKind), step(ClusterServiceVersionKind), step(crdKind), step(crdKind)},
+		},
+		{
+			description: "csvsCRDSAndRandomKinds",
+			in:          []*Step{step(crdKind), step(ClusterServiceVersionKind), step(crdKind), step(crdKind), step("These"), step("are"), step("random"), step("Kinds")},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.description, func(t *testing.T) {
+			result := OrderSteps(tt.in)
+			require.EqualValues(t, len(result), len(tt.in))
+			require.True(t, isOrdered(result))
+		})
+	}
+}
+
+func step(kind string) *Step {
+	resource := StepResource{}
+	resource.Kind = kind
+
+	result := &Step{}
+	result.Resource = resource
+	return result
+}
+
+func isOrdered(steps []*Step) bool {
+	var csvSeen, otherResourceSeen bool
+	for _, step := range steps {
+		switch step.Resource.Kind {
+		case crdKind:
+			if csvSeen || otherResourceSeen {
+				return false
+			}
+		case ClusterServiceVersionKind:
```

Comment:
shouldn't we set `csvSeen` to true in this case? right now this function always returns true